src/components/inputs/money-input/money-input.spec.js

```diff
@@ -508,65 +491,65 @@ describe('rendering', () => {
-          .findWhere(item => item.prop('children') === 'USD')
-          .prop('onClick')({ target: { value: '12' } });
-      });
-
-      it('should call onChange with an event', async () => {
-        // wait for onChange call which happens in setTimeout(fn ,0)
-        // This can be removed once the setTimeout workarounds are
-        // removed from money-input.
-        await new Promise(resolve => setTimeout(resolve, 1));
-        expect(props.onChange).toHaveBeenCalledWith(
-          expect.objectContaining({
-            persist: expect.any(Function),
-            target: expect.objectContaining({
-              name: `foo.currencyCode`,
-              value: 'USD',
-            }),
-          })
-        );
-      });
-    });
-  });
+  // describe('currency field', () => {
+  //   describe('when selecting the already used currency', () => {
+  //     beforeEach(() => {
+  //       props = createTestProps();
+  //       wrapper = shallow(<MoneyInput {...props} />);
+
+  //       dowshiftRenderWrapper = wrapper
+  //         .find(CurrencyDropdown)
+  //         .dive()
+  //         .renderProp('children', { isOpen: true, toggleMenu: jest.fn() });
+
+  //       dowshiftRenderWrapper
+  //         .find(Option)
+  //         // click the currency which is already selected
+  //         .findWhere(item => item.prop('children') === props.value.currencyCode)
+  //         .prop('onClick')({ target: { value: '12' } });
+  //     });
+
+  //     it('should not call onChange', () => {
+  //       expect(props.onChange).not.toHaveBeenCalled();
+  //     });
+  //   });
+  //   describe('when changing currency', () => {
+  //     beforeEach(() => {
+  //       props = createTestProps({ name: 'foo' });
+  //       wrapper = shallow(<MoneyInput {...props} />);
+
+  //       dowshiftRenderWrapper = wrapper
+  //         .find(CurrencyDropdown)
+  //         .dive()
+  //         .renderProp('children', { isOpen: true, toggleMenu: jest.fn() });
+
+  //       dowshiftRenderWrapper
+  //         .find(Option)
+  //         // clicking the already selected one (EUR) won't trigger an action,
+  //         // so we click the second one (USD)
+  //         .findWhere(item => item.prop('children') === 'USD')
+  //         .prop('onClick')({ target: { value: '12' } });
+  //     });
+
+  //     it('should call onChange with an event', async () => {
+  //       // wait for onChange call which happens in setTimeout(fn ,0)
+  //       // This can be removed once the setTimeout workarounds are
+  //       // removed from money-input.
+  //       await new Promise(resolve => setTimeout(resolve, 1));
+  //       expect(props.onChange).toHaveBeenCalledWith(
+  //         expect.objectContaining({
+  //           persist: expect.any(Function),
+  //           target: expect.objectContaining({
+  //             name: `foo.currencyCode`,
+  //             value: 'USD',
+  //           }),
+  //         })
+  //       );
+  //     });
+  //   });
+  // });
```

Comment:
This is WIP?

Reply:
Hehe no worries :) Better eager than late/never :P I already refactored these tests to `react-testing-library` on another branch (for which a PR is open). I didn't merge anything today due to GitHub being flaky. I need to merge my open branches in the proper order, then rebase and then these tests might actually only need few changes.

Reply:
Same here. I do not dare to merge anything almost.